tensorflow_io/arrow/kernels/arrow_dataset_ops.cc

```diff
@@ -871,13 +864,51 @@ class ArrowFeatherDatasetOp : public ArrowOpKernelBase {
   };
 };
 
+
+enum ArrowHostType {
+  AF_INET,
+  AF_UNIX,
+  STDIN,
+};
+
+Status GetHostTypeStr(ArrowHostType host_type, string* host_type_str) {
+  switch (host_type) {
+    case ArrowHostType::AF_INET:
+      *host_type_str = "AF_INET";
+      break;
+    case ArrowHostType::AF_UNIX:
+      *host_type_str = "AF_UNIX";
+      break;
+    case ArrowHostType::STDIN:
+      *host_type_str = "STDIN";
+      break;
+    default:
+      return errors::Internal("Unsupported host_type: " +
+                              std::to_string(host_type));
+  }
+  return Status::OK();
+}
+
+Status GetHostType(string host_type_str, ArrowHostType* host_type ) {
+  if (host_type_str == "AF_INET") {
+    *host_type = ArrowHostType::AF_INET;
+  } else if (host_type_str == "AF_UNIX") {
+    *host_type = ArrowHostType::AF_UNIX;
+  } else if (host_type_str == "STDIN") {
+    *host_type = ArrowHostType::STDIN;
+  } else {
+    return errors::Internal("Unsupported host type: " + host_type_str);
+  }
+  return Status::OK();
+}
```

Comment:
This enum conversion is kind of messy, it could be improved in the future I think.